dimod/binary_quadratic_model.py

```diff
@@ -2048,6 +2048,43 @@ def from_qubo(cls, Q, offset=0.0):
 
         return cls(linear, quadratic, offset, Vartype.BINARY)
 
+    @classmethod
+    def from_markov_random_field(cls, edges):
+        """
+
+
+        Examples:
+            >>> fixed = {'n01': 1, 'n03': 0. }
+            >>> edges = [['n01', 'n02', 3.0, 4.0, -6.0, 2.0], 
+                         ['n02', 'n04', 5.0, -1.0, -4.0, 2.0], 
+                         ['n03', 'n04', 1.0, 6.0, -6.0, 2.0], 
+                         ['n03', 'n05', -4.0, -3.0, 6.0, 5.0]]
+            ...
+            >>> # generate the markov network in a form that can be solved
+            >>> network = dimod.BinaryQuadraticModel.from_markov_random_field(edges)
+            ...
+            >>> # next we want to clamp/fix some of the variables
+            >>> clamped = network.copy()
```

Comment:
Why `copy`?

Reply:
`fix_variables` acts on the BQM in-place, so we need to copy it first if we want to keep the original bqm (in this case it doesn't matter)